src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,167 @@
+     */
+    public function sourceRecords($params)
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                $params['Count'] = $desiredCount++;
+            }
+        }
+
+        $members = $this->applyParams(Member::get(), $params)->toArray();
+        array_walk($members, function ($member) use ($methods) {
+            $defaultMethod = $member->getDefaultRegisteredMethod();
+            if ($defaultMethod) {
+                $defaultMethodClassName = $defaultMethod ? $defaultMethod->MethodClassName : null;
+                $backupClass = get_class(Injector::inst()->get(MethodRegistry::class)->getBackupMethod());
+                $registeredMethods = $member
+                    ->RegisteredMFAMethods()
+                    ->exclude('MethodClassName', $backupClass)
+                    ->column('MethodClassName');
+                $registeredMethodNames = array_map(function ($methodClass) use ($methods) {
+                    return $methods[$methodClass];
+                }, $registeredMethods);
+                $registeredMethodList = implode(', ', $registeredMethodNames);
+                $methodCount = count($registeredMethods);
+                $member->methodCount = $methodCount ?: '';
+                $member->registeredMethods = $registeredMethodList;
+                $member->defaultMethodName = $methods[$defaultMethodClassName] ?? '';
+            }
+            $member->skippedRegistration = $member->HasSkippedMFARegistration
+                ? $member->dbObject('HasSkippedMFARegistration')->Nice()
+                : '';
+        });
+        $filteredList = ArrayList::create($members);
+        if (isset($countFilter)) {
+            $filteredList = $filteredList->filter('methodCount', $countFilter);
+        }
+        return $filteredList;
+    }
+
+    public function columns()
+    {
+        return singleton(Member::class)->summaryFields() + [
+            'methodCount' => _t(__CLASS__ . '.COLUMN_METHOD_COUNT', 'Number of registered methods'),
+            'registeredMethods' => _t(__CLASS__ . '.COLUMN_METHODS_REGISTERED', 'Registered methods'),
+            'defaultMethodName' => _t(__CLASS__ . '.COLUMN_METHOD_DEFAULT', 'Default method'),
+            'skippedRegistration' => _t(__CLASS__ . '.COLUMN_SKIPPED_REGISTRATION', 'Skipped registration'),
+        ];
+    }
+
+    public function parameterFields()
```

Comment:
Needs a docblock. At least I don't think this is a method on reports?

Reply:
It is invisible API.

Reply:
Huh?

Reply:
There is no method on Report, however Report does explicitly search for and call `parameterFields` (from within `getCMSFields` iirc).
The docblock also makes reference to it https://github.com/silverstripe/silverstripe-reports/blob/b5232aa7ab4d5a1151252fa396bbfd90e01dd0db/code/Report.php#L47